omod/src/main/java/org/openmrs/module/owa/activator/OwaActivator.java

```diff
@@ -163,4 +172,32 @@ public void stopped() {
 		log.info("OWA Module stopped");
 	}
 	
+	/**
+	 * Used to set the login Url Global Property that directs to the Add-on OWA UI if no UI module
+	 * is installed. Searches against the legacy and Reference Application module on installed UI modules.
+	 */
+	private void configureLoginUrl(AppManager appManager) {
+		if (appManager.exists(ADD_ON_MANAGER_APP)) {
+			if (!uiModuleInstalled()) {
+				String folderName = null;
+				for (App app : appManager.getApps()) {
+					if (app.getName().equalsIgnoreCase(ADD_ON_MANAGER_APP)) {
+						folderName = app.getFolderName();
+					}
+				}
+				Context.getAdministrationService().setGlobalProperty(LOGIN_URL, "owa/" + folderName + "/index.html");
+				return;
+			}
+		}
+		Context.getAdministrationService().setGlobalProperty(LOGIN_URL, "login.htm");
+	}
+	
+	private Boolean uiModuleInstalled() {
```

Comment:
Is the assumption that this functionality works with only the legacyui and referenceapplication modules, but nothing else?

Reply:
Part of the design discussion we've had with @bmamlin, has been to use this approach for the few currently used ui modules that provide for login, and that a more generic approach will be developed to handle usage of any user login module available as need arises. Partly to the suggestions made was have a GP string that's updated with installed UI modules. Consulting this list is more generic than having to hard code them the way it is done here.